modules/uhttp/uhttpclient/client.go

```diff
@@ -25,14 +25,23 @@ import (
 	"time"
 
 	"go.uber.org/fx/auth"
+
+	"github.com/opentracing/opentracing-go"
 )
 
 // New creates an http.Client that includes 2 extra outbound middleware: tracing and auth
 // they are going to be applied in following order: tracing, auth, remaining outbound middleware
 // and only if all of them passed the request is going to be send.
 // Client is safe to use by multiple go routines, if global tracer is not changed.
-func New(info auth.CreateAuthInfo, middleware ...OutboundMiddleware) *http.Client {
-	defaultMiddleware := []OutboundMiddleware{tracingOutbound(), authenticationOutbound(info)}
+func New(tracer opentracing.Tracer, info auth.CreateAuthInfo, middleware ...OutboundMiddleware) *http.Client {
```

Comment:
I think CreateAuthInfo was meant to be a collection of the base objects. It has auth and metrics right now. Either add tracer to it (and rename interface) or make all of them separate arguments.